src/pages/NewsPage/index.tsx

```diff
@@ -1,117 +1,168 @@
+
+	if (isLoading) {
+		return <div>Loading...</div>;
+	}
+
+	if (isError) {
+		return <div>Error: Failed to fetch news</div>;
+	}
+
+	return (	 
+		<div>
+		{/* eslint-disable-next-line */}
+		{data && data?.metadata.map((article: NewsItemProps) => (
+			<NewsItem
+			key={article.url}
+			title={article.title}
+			urlToImage={article.urlToImage}
+			url={article.url}
+			publishedAt={article.publishedAt}
+			/>
+		))}
+		</div>
+	);
+};
+  
+
 export default function NewsPage() {
-	const navigate = useNavigate();
-	const handleNavigateToDashboardPage = () => {
-		navigate("/dashboard");
+	const [selectedCategory, setSelectedCategory] = useState("drug");
+
+	const handleTabChange = (category: string) => {
+		setSelectedCategory(category);
 	};
+
 	return (
+		<QueryClientProvider client={queryClient}>
 		<div className="min-h-screen">
-			<div className="mb-12 mt-24 flex flex-col gap-4">
-				<div className="flex justify-center relative">
-					<IoIosArrowBack
-						className="text-3xl absolute left-0"
-						onClick={handleNavigateToDashboardPage}
-					/>
-					<header className="text-3xl font-bold text-center">
-						News
-					</header>
-				</div>
-				<div className="flex flex-row gap-10 items-center justify-center">
-					<Button title="Drugs" imageClassName="h-[105%] w-[105%]" />
-					<Button
-						title="Cyber Bully"
-						imageClassName="h-[105%] w-[105%]"
+			<div className="pb-12 pt-8 xl:pb-12 xl:pt-24 flex flex-col gap-8">
+				<div className="text-3xl xl:text-5xl font-bold text-center xl:text-center">News</div>
+				<div className="flex flex-row gap-2 xl:gap-10 justify-center">
+					<Button 
+						title="Drugs" 
+						onClick={() => handleTabChange("drug")} 
 					/>
-					<Button title="Vape" imageClassName="h-[105%] w-[105%]" />
-					<Button
-						title="Heritage"
-						imageClassName="h-[105%] w-[105%]"
+					<Button 
+						title="Vape" 
+						onClick={() => handleTabChange("vape")} 
 					/>
 				</div>
-				<div className="flex flex-col gap-4 items-center">
-					<NewsItem
-						thumbnail={newsThumbnail}
-						title="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Labore, optio consequuntur."
-						newsLink="https://example.com/news"
-					/>
-					<NewsItem
-						thumbnail={newsThumbnail}
-						title="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Labore, optio consequuntur."
-						newsLink="https://example.com/news"
-					/>
-					<NewsItem
-						thumbnail={newsThumbnail}
-						title="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Labore, optio consequuntur."
-						newsLink="https://example.com/news"
-					/>
-					<NewsItem
-						thumbnail={newsThumbnail}
-						title="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Labore, optio consequuntur."
-						newsLink="https://example.com/news"
-					/>
+				<div className="flex flex-col gap-10 item-center">
+					{selectedCategory === "drug" && <NewsComponentDrug />}

```

Comment:
i think this can be improved by using searchParams. Maybe u can take alook at PuzzlePage/index.tsx . Do this if you have the time